### PR TITLE
Keycloak account console PatternFly design system alignment

### LINKED ISSUE #11103: Keycloak account console PatternFly design system alignment
### Description

The existing account console theme uses an outdated version of the PatternFly library. Work has been completed to update the Keycloak Account Console to the latest version of the PatternFly design system (release 2021.15) and to follow the latest PatternFly design guidelines. This also brings the account console into visual / design alignment with the latest updates to the admin console.

[Discussion](https://github.com/keycloak/keycloak/discussions/9923) and [pull request](https://github.com/keycloak/keycloak/pull/10748). 

### Discussion

https://github.com/keycloak/keycloak/discussions/9923

### Motivation

_No response_

### Details

The most obvious visual changes applied to all pages are:

- Page content areas use a white background rather than white cards on a grey background (except for landing page)
- Left aligning content
- Updating font to Red Hat Text (body copy) and Red Hat Display (headings)
- Using sentence case for all text
  
Global code changes include: 

- @patternfly/react libraries updated to latest versions
- Fixes for breaking changes from the library updates 
- Layout and styling updates to accommodate designs

### PR DESCRIPTION
**Pull request by tyandor**
This work updates the Keycloak Account Console to the latest version of the PatternFly design system (release 2021.15) and follows the latest PatternFly design guidelines. This also brings the account console into visual / design alignment with the latest updates to the admin console.

See the design proposal for more details.

Closes #11103

### CODE REVIEW

**Comment by tyandor** [comment]
@ssilvert @jonkoops @edewit @vmuzikar apologies for the inconvenience, but I had to recreate this. The commits are now accurate to just this PR and build should be working.

**Comment by edewit** [comment]
so I also need to recreate my review... or can you see the things that I put on the other PR?

**Comment by edewit** [comment]
on the card page the "hamburger" icon and the signout button are both visible:
image

**Comment by tyandor** [comment]
> so I also need to recreate my review... or can you see the things that I put on the other PR?

@edewit I can still see the closed PR

**Comment by edewit** [comment]
@tyandor okay recreated the review comments anyway

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx:501]: It's not allowed to have both an `aria-labelledby` and `aria-label` attribute, they are mutually exclusive. Also an `aria-label` attribute has to be a human readable text, as it's what the screen reader will read to the user.

**Review by edewit — CHANGES_REQUESTED** [review]

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx:192]: Some commented out code here.

File: themes/src/main/resources/theme/keycloak.v2/account/resources/public/patternfly-addons.css
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/resources/public/patternfly-addons.css:1]: This seems like a style sheet that should be loaded from the PatternFly package, can we load this in from there?

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx:68]: Some commented out code.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/PageHeaderTool.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/PageHeaderTool.tsx:9]: This could be a function component instead, class components are on the way out.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx:79]: Not a fan of leaving code commented out like this, can we remove it instead?

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx:69]: Text should be translated.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx:63]: Where is the element with the `id` "refresh page"?

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/account-page/AccountPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/account-page/AccountPage.tsx:167]: Text should be translated.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx:190]: Some commented out code

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx:196]: Some commented out code

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx:211]: Text should be translated.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx:94]: Should be translated.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionRequest.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionRequest.tsx:52]: Can you add a correct type her instead of `any`?

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx:49]: Can we add the correct type here?

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx:200]: Aria label should be a human readable and translated string.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx:49]: @edewit this was just fixing an implicit any error coming from the original code. I don't have access to a resource server, do you know what this interface needs to be?

**Review by stianst — COMMENTED** [review]
One thing to bear in mind is that this will be a significant change for anyone that customises the account console, so we will need an entry in the migration guide.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx
Comment (edewit) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx:49]: it should be `{permissions: Permission[]}`

**Review by ssilvert — CHANGES_REQUESTED** [review]
I won't repeat the thorough code review done by Erik and Jon, but I have manually tested this and overall it looks good.

The only issue I found was the same thing Erik found where he says, "on the card page the "hamburger" icon and the signout button are both visible".  I think he is actually talking about the kebab menu.  The kebab should not be shown in large screen mode.  The way this works is that when screen is large, we show "Back to <application>" and "Sign out" button.  When screen is small we hide those and show kebab menu instead.

Alternatively, I'd be ok if we just got rid of the kebab menu altogether.  Right now I don't see it in the application, but only see it on the home page.  

In the application "Back to <application>" appears left justified and butts up against the Keycloak logo when screen is small.  So if we do get rid of the kebab menu, that should be addressed.

I have manually tested everything except "Resources".  @tyandor have you tested this section?  If not, one of use needs to go through the trouble of testing it.

As for automated testing, it's such a pain to run our Arquillian tests for this, I'm leaning against making you do it.  They will probably fail and they need to be rewritten in Cypress anyway.

So as I see it, to merge we need:

1. Address code changes from Erik and Jon.
2. Fix kebab issue
3. Update migration guide
4. Verify that Resources section works.  If nobody has done that yet, I can take care of it.

**Comment by tyandor** [comment]
@ssilvert sounds good! I'm working through just a couple things from earlier feedback, and will take another shot at setting up a resource server today. We ran into issues with that previously following the docs and book, so might need you to validate.

File: themes/src/main/resources/theme/keycloak.v2/account/src/tsconfig.json
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/tsconfig.json:15]: @jonkoops there are several errors caused by newer versions of @react/react-core around the way they handle imports. We have an issue created on the pf repo to fix some of the bundler problems caused by the setup there, but for now I think we need this and don't see it causing any problems.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/widgets/ContinueCancelModal.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/widgets/ContinueCancelModal.tsx:96]: This was requested by UX designers as alignment with PF

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx:501]: So oddly both are required on the DataListAction in the DataList component from PF. In which case, yeah the `labeledby` will take precedence. I've updated all these to be translatable, screen reader readable, strings.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/PageHeaderTool.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/PageHeaderTool.tsx:9]: I think that could be said of this entire theme :) We discussed refactoring all of these, but it was determined to be beyond the scope of the current task.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx:79]: Yes! Cleaning it up

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx:63]: This should be `aria-label`...adding that and adding localization

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx]: This is converted to switch

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx]: That didn't do the trick, but adding a new type for Permissions I was able to fix this and another similar instance.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/package.json
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/package.json:15]: Addons has been moved into this and ref updated in index.ftl.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/widgets/EmptyMessageState.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/widgets/EmptyMessageState.tsx]: Yup, resolved 👍

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx]: Removed

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/signingin-page/SigningInPage.tsx]: Removed

File: themes/src/main/resources/theme/keycloak.v2/account/resources/public/patternfly-addons.css
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/resources/public/patternfly-addons.css]: Yes, I moved this into the pf copy scripts and removed the file.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/App.tsx]: Removed

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/ContentPage.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/account-page/AccountPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/account-page/AccountPage.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Removed

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Removed

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/ShareTheResource.tsx]: Resolved, along with a few other aria labels on this page.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionSelect.tsx]: Resolved

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionRequest.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/PermissionRequest.tsx]: New type created for Permissions, used here. and in one additional instance.

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/my-resources-page/EditTheResource.tsx]: Added

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx
Comment (tyandor) [reply themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx]: Fixed

**Comment by tyandor** [comment]
@ssilvert @jonkoops @edewit I've completed all the fixes and updates if you want to verify. Assuming the process is for y'all to validate issues are resolved, so I just added comments. 

Stan, are you able to validate Resources? I still haven't been able to get that set up. And any guidance on what is involved in updating the migration guide would be helpful.

**Review by edewit — DISMISSED** [review]
:+1:

**Comment by ssilvert** [comment]
> 
> Stan, are you able to validate Resources? I still haven't been able to get that set up. And any guidance on what is involved in updating the migration guide would be helpful.

Getting resource stuff set up is a beast.  I finally got the photoz example working well enough to test account console and it looks good.  I also verified the issue with the kabab is resolved.

**Review by ssilvert — DISMISSED** [review]

**Comment by tyandor** [comment]
Thanks @ssilvert! We'll probably need someone from docs to help with the update there. At this point is there anything preventing this from being merged?

**Comment by ssilvert** [comment]
> Thanks @ssilvert! We'll probably need someone from docs to help with the update there. At this point is there anything preventing this from being merged?

@tyandor I think you will need to update the document yourself.  Our docs guy can review if you want.

@stianst I think the code is ready, but I'll leave it up to you as to whether we can merge before the migration guide is updated.  Since it's close to release time, I'm not sure if you want to hold off or not.

**Review by jonkoops — DISMISSED** [review]

**Comment by stianst** [comment]
Would be good to get commits squashed for this one, and there's also a large number of test failures that needs to be resolved before we can merge.

Do we have a GitHub issue for this one?

**Comment by tyandor** [comment]
> Would be good to get commits squashed for this one, and there's also a large number of test failures that needs to be resolved before we can merge.
> 
> Do we have a GitHub issue for this one?
> 
> If we want this to be included in Keycloak 18 we need the above sorted out asap.

@stianst the commits have been squashed and the test failures should be resolved (all succeeded for me locally). 

Here is the [issue](https://github.com/keycloak/keycloak/issues/11103).

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx:223]: Stray console statement.
```suggestion
```

File: themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx
Comment (jonkoops) [inline themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx:224]: This code cannot be right, none of the cases besides the `default` will ever match.

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**
